### PR TITLE
awscli2: 2.7.33 -> 2.8.5

### PR DESCRIPTION
**Pull request by anthonyroussel**
###### Description of changes

* Upgrade awscli2 from 2.7.33 to 2.8.5
  * Migrate build system to pyproject / flit
  * Removed useless overrides in pyproject.toml (was setup.cfg)
  * Enabled tests execution (were disabled by checkPhase)
    * Disable xdist (test crashes with OOM error on my laptop)
    * Override prompt-toolkit to 3.0.28 (3.0.29 has a breaking change which broke cli-auto-prompt feature)
  * See https://github.com/aws/aws-cli/blob/2.8.5/CHANGELOG.rst#L5

* Added myself to maintainers

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by devusb** [comment]
Result of `nixpkgs-review pr 197270` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>awscli2</li>
    <li>liquidsoap</li>
  </ul>
</details>

**Comment by maxeaubrey** [comment]
Thank you for doing the work on this. I'm a little concerned with how slow tests are - I made the mistake of trying to build on my laptop the first time, and only moving to a 32 thread Ryzen made this a semi-tolerable experience (at "just" ~10 minutes). I suppose as long as there's a cached build this is fine, but after 20 minutes I was (according to the output) 1% done on my 8 thread intel laptop.

**Comment by devusb** [comment]
Result of `nixpkgs-review pr 197270` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>liquidsoap</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>awscli2</li>
  </ul>
</details>

**Comment by anthonyroussel** [comment]
> Thank you for doing the work on this. I'm a little concerned with how slow tests are - I made the mistake of trying to build on my laptop the first time, and only moving to a 32 thread Ryzen made this a semi-tolerable experience (at "just" ~10 minutes). I suppose as long as there's a cached build this is fine, but after 20 minutes I was (according to the output) 1% done on my 8 thread intel laptop.

Hello
Thanks for the review :)

Yes you're right.
These tests are so slow, it took at least 10/15 minutes to run on my laptop.

I will take a look if it's possible to skip some "slow tests" in the AWS CLI PyTest suite.

**Comment by anthonyroussel** [comment]
I disabled the integration, backend and functional tests (were very slow >= 10 min).

I only kept the unit tests execution to ensure that the program "is mostly tested" without degrading the build time.

**Review by bryanasdev000 — APPROVED** [review]
Welcome aboard!

Diff and ofborg LGTM, thanks :D

File: pkgs/tools/admin/awscli2/default.nix [hunk 1]
```diff
@@ -87,13 +92,47 @@ with py.pkgs; buildPythonApplication rec {
...
+
+  disabledTestPaths = [
+    # Integration tests require networking
+    "tests/integration"
+
+    # Disable slow tests (only run unit tests)
+    "tests/backends"
+    "tests/functional"
+
+    # Disable failing tests
```
[the remark this answers is not in the document]
Comment (maxeaubrey) [reply pkgs/tools/admin/awscli2/default.nix:114]: Adding a comment explaining the specifics of these failures would be appreciated. I think after that I can hit merge.

[the remark this answers is not in the document]
Comment (anthonyroussel) [reply pkgs/tools/admin/awscli2/default.nix]: > Adding a comment explaining the specifics of these failures would be appreciated. I think after that I can hit merge.

It's strange, looks like rebasing my branch against an up-to-date master branch was a resolution to these faulty tests on my laptop.

So I just removed these disabled test paths.
And the build is now ok :)

**Pull request merged**